### PR TITLE
Adding chunkSize to the HTTP config

### PR DESCRIPTION
**Pull request by msmakouz**
| Q             | A
| ------------- | ---
| Bugfix?       | ❌
| Breaks BC?    | ❌ 
| New feature?  | ✔️

### CODE REVIEW

File: src/Http/src/Config/HttpConfig.php [hunk 1]
```diff
@@ -51,4 +52,12 @@ public function getMiddleware(): array
     {
         return $this->config['middleware'] ?? $this->config['middlewares'];
     }
+
+    /**
+     * If chunkSize isn't provided - using default values
+     */
+    public function getChunkSize(): ?int
+    {
+        return !\is_null($this->config['chunkSize']) ? (int) $this->config['chunkSize'] : null;
```
Comment (roxblnfk) [inline src/Http/src/Config/HttpConfig.php:61]: returned value should be positive int or null

File: src/Framework/Bootloader/Http/HttpBootloader.php [hunk 2]
```diff
@@ -122,4 +123,18 @@ protected function httpCore(
 
         return $core;
     }
+
+    /**
+     * @noRector RemoveUnusedPrivateMethodRector
+     */
+    private function createEmitter(HttpConfig $config): EmitterInterface
```
Comment (roxblnfk) [inline src/Framework/Bootloader/Http/HttpBootloader.php:130]: mb just do it public?

File: src/Http/src/Config/HttpConfig.php (hunk 1 above)
Comment (msmakouz) [reply src/Http/src/Config/HttpConfig.php]: Added a check that the value is greater than 0

File: src/Framework/Bootloader/Http/HttpBootloader.php (hunk 2 above)
Comment (msmakouz) [reply src/Framework/Bootloader/Http/HttpBootloader.php]: Done

File: src/Http/src/Config/HttpConfig.php [hunk 3]
```diff
@@ -55,9 +55,15 @@ public function getMiddleware(): array
 
     /**
      * If chunkSize isn't provided - using default values
+     *
+     * @psalm-return positive-int
      */
     public function getChunkSize(): ?int
     {
-        return !\is_null($this->config['chunkSize']) ? (int) $this->config['chunkSize'] : null;
+        if (\is_null($this->config['chunkSize']) || (int) $this->config['chunkSize'] <= 0) {
```
Comment (roxblnfk) [inline src/Http/src/Config/HttpConfig.php:63]:
```suggestion
        if (\is_null($this->config['chunkSize']) || (int) $this->config['chunkSize'] < 0) {
```

I propose to support `0` value. We need to update SapiEmitter also:
```php
// SapiEmitter::emitBody
if ($this->bufferSize === 0) {
    return (string) $body;
}
while (!$body->eof()) {
// ...
```
And need to add comment for the `SapiEmitter::$bufferSize` property about the `0` value that means disabled streamed response

Comment (msmakouz) [reply src/Http/src/Config/HttpConfig.php]: Done.

**Review by roxblnfk — APPROVED** [review]

File: src/Http/src/Config/HttpConfig.php [hunk 4]
```diff
@@ -51,4 +52,16 @@ public function getMiddleware(): array
...
+
+    /**
+     * If chunkSize isn't provided - using default values
+     */
+    public function getChunkSize(): ?int
+    {
+        if (\is_null($this->config['chunkSize']) || (int) $this->config['chunkSize'] < 0) {
+            return null;
+        }
+
```
Comment (butschster) [inline src/Http/src/Config/HttpConfig.php:64]: Maybe do you replace `(int) $this->config['chunkSize'] < 0` with `assert((int) $this->config['chunkSize'] > 0, 'Chunk size should be greater than 0.')` ?

**Review by butschster — CHANGES_REQUESTED** [review]

File: src/Http/src/Config/HttpConfig.php (hunk 4 above)
Comment (roxblnfk) [reply src/Http/src/Config/HttpConfig.php:64]: Chunk size can be equal 0

**Pull request merged**